blip-00xx.md

```diff
@@ -0,0 +1,220 @@
+  * [`u32`:`fee_proportional_millionths`]
+  * [`u16`:`cltv_expiry_delta`]
+  * [`u64`:`htlc_minimum_msat`]
+  * [`u64`:`htlc_maximum_msat`]
+  * [`u16`:`flen`]
+  * [`flen*byte`:`features`]
+  * [`33*byte`:`first_ephemeral_blinding_point`]
+  * [`byte`:`num_hops`]
+  * [`num_hops*blinded_hop`:`blinded_hops`]
+
+1. subtype: `blinded_hop`
+2. data:
+   * [`33*byte`:`blinded_node_pubkey`]
+   * [`2*byte`:`cipher_text_length`]
+   * [`cipher_text_length*byte`:`cipher_text`]
+
+The `blinded_node_pubkey` of the first `blinded_hop` in a `blinded_payinfo` is
+the real public key of the blinded path's introduction node. 
+
+### Requirements
+
+An invoice containing the `b` field type:
+- MUST not contain the `r` field type.
+- MUST not contain the `s` field type since a payment address in the context of
+  blinded payments does not make sense since the recipient is able to use the
+  `path_id` in the `encrypted_recipient_data` for the same purpose.
+- SHOULD sign the invoice with a private key that is not the same as their 
+  public node ID and should not set the destination node (`n` field).
+
+### Size Restrictions
+
+#### `data_length` Limit
+
+In order to conform to any existing BOLT 11 invoice parser, each new tagged 
+field must use the `data_length` encoding defined there. This means that the 
+maximum size of any _single_ encoded blinded path is 639 bytes. 
+
+What follows is a rough estimation of the maximum number of hops we can include 
+in a single blinded path. It assumes that each hop's cipher text is the same 
+length. 
+
+##### Cipher Text Size Estimation
+
+First, a rough estimation of the average cipher text length is required. A 
+forwarding node in a blinded path will receive a cipher text payload containing
+the following data:
+
+- `padding`: optional
+- `short_channel_id` of 8 bytes
+- `payment_relay`: 
+    * 2 byte `cltv_expiry_delta`
+    * 4 byte `fee_proportional_millionths`
+    * 4 byte `fee_base_msat`
+- `payment_constraints`:
+    * 4 byte `max_cltv_expiry`
+    * 8 byte `htlc_minimum_msat`
+- `allowed_features`: optional
+
+If we assume that the `allowed_features` vector is not set, then this comes to 
+a total of 30 mandatory bytes. 
+
+For the recipient node, it will receive a cipher text payload containing:
+
+- `padding`: optional
+- `path_id`: let's assume that this is 32 bytes like the existing payment 
+   address.
+- `payment_constraints`:
+    * 4 byte `max_cltv_expiry`
+    * 8 byte `htlc_minimum_msat`
+
+This comes to a total of 44 bytes for the recipient's cipher text. 
+
+The padding field should be used by recipients to pad cipher text blobs so that 
+all are the same size. Since the calculated recipient cipher text blob size (44) 
+is larger than that of the forwarding nodes (30), we can assume that all the
+cipher text blobs will have a size of around 44 bytes. 
+
+##### `blinded_hop` Size Estimation
+
+The total number of bytes required for a single `blinded_hop` is:
+
+    = 33+2+len(cipher_text)
```

Comment:
don't we need to add the padding value here or is it already included in the cipher_text.

Reply:
see the `##### Cipher Text Size Estimation` section above - the cipher text will already contain the padding